src/components/letter/representative-apis.tsx

```diff
@@ -0,0 +1,275 @@
+import { makeAddressLine } from "./utils";
+import {
+  BlackmadCityCountilResponse,
+  OfficialRestrict,
+  GoogleCivicRepsResponse,
+  OfficialAddress,
+  LevelsAndRoles,
+} from "./types";
+import { getGeocode, getLatLng } from "use-places-autocomplete";
+import _ from "lodash";
+
+export const mungeCityCouncil = (
+  cityCouncilMembers: BlackmadCityCountilResponse,
+  restricts?: OfficialRestrict[]
+): OfficialAddress[] => {
+  if (
+    !cityCouncilMembers ||
+    !cityCouncilMembers.data ||
+    cityCouncilMembers.data.length === 0
+  ) {
+    return [];
+  }
+
+  return _.flatMap(cityCouncilMembers.data, (cityCouncilMember) => {
+    // only use restricts if present, otherwise let everything through
+    if (
+      !_.some(
+        restricts || [],
+        (restrict) =>
+          restrict.level === cityCouncilMember.office.level &&
+          (restrict.role === cityCouncilMember.office.role ||
+            (restrict.role as string) === "*")
+      )
+    ) {
+      return [];
+    }
+
+    let link: string;
+    if (cityCouncilMember.body === "New York City Council") {
+      link = `https://defund-nypd-reps.glitch.me/district/${cityCouncilMember.district}`;
+    } else {
+      link = cityCouncilMember.urls?.[0];
+    }
+
+    return cityCouncilMember.addresses.map((address) => {
+      const officeName = address.name
+        ? cityCouncilMember.office.name + " - " + address.name
+        : cityCouncilMember.office.name;
+      return {
+        address: {
+          name: cityCouncilMember.name,
```

Comment:
maybe we can construct these rather than building them as literals?

Reply:
I'm not sure what you mean / what the benefit would be.

We could have a class that had a constructor that assembled these bits into an OfficialAddress but that's basically what this function does. To make such a constructor safe and easy to use, we'd give it a big named object argument which would look pretty much exactly like what's happening here. I think it would just be extra noise.

It's typechecked all the way through as an OfficalAddress because that's the return type - if this was missing a field from OfficialAddress it would be a compile error.

Reply:
renamed everything to be slightly less terribly named